src/main.rs

```diff
@@ -173,7 +164,34 @@ fn before_each_package<P: AsRef<Path>>(
             std::io::ErrorKind::Other,
             format!("Unknown/unsupported package format {_other:?}"),
         )),
+    };
+
+    // Only proceed with resource copying if compilation succeeded
+    compilation_result?;
```

Comment:
i understand this line, but it's a pretty strange way to write this.

It'd be clearer for future readers if you just added a `?` to the end of each call to `before_each_package_*` above, and removed these 2 lines and the `let compilation_result = `.